lib/styleguide.test.js

```diff
@@ -0,0 +1,126 @@
+        'blockquote.css',
+        'video.css'
+      ],
+      vultureMockData = [
+        'blockquote.css',
+        'blockquote_red.css',
+        'blockquote_blue.css',
+        'video.css',
+        'video_fullbleed.css',
+        'video_mobile.css'
+      ];
+
+    sandbox = sinon.sandbox.create();
+    sandbox.stub(files);
+
+    files.getFiles.withArgs('styleguides/di/components').returns(diMockData);
+    files.getFiles.withArgs('styleguides/vulture/components').returns(vultureMockData);
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  it('get all available component variations', function () {
+    var expected = {
+      blockquote: [ 'blockquote_red', 'blockquote_blue' ],
+      video: [ 'video_fullbleed', 'video_mobile' ]
+    };
+
+    expect(lib.getVariations('vulture')).to.deep.equal(expected);
+  });
+
+  it('if there is no styleguide set, there should be no variations set', function () {
+    expect(lib.getVariations()).to.deep.equal({});
+  });
+
+  it('if there are no component variations, return an empty object', function () {
+
+    expect(lib.getVariations('di')).to.deep.equal({});
+  });
+
+  it('set default variations for components that have variations but none set', function () {
+    var testData = [
+      {
+        _ref: 'www.vulture.com/_components/blockquote/instances/foo@published',
+        componentVariation: 'blockquote_red'
+      },
+      {
+        _ref: 'www.vulture.com/_components/blockquote/instances/bar@published',
+      },
+      {
+        _ref: 'www.vulture.com/_components/clay-paragraph/instances/cjeq5662h001b3i61rr8fffu1@published',
+      },
+      {
+        _ref: 'www.vulture.com/_components/article-sidebar/instances/foobar@published',
+        content: [
+          { _ref: 'www.vulture.com/_components/blockquote/instances/pho@published'}
+        ]
+      }
+      ],
+      expectedData = [
+        {
+          _ref: 'www.vulture.com/_components/blockquote/instances/foo@published',
+          componentVariation: 'blockquote_red'
+        },
+        {
+          _ref: 'www.vulture.com/_components/blockquote/instances/bar@published',
+          componentVariation: 'blockquote'
+        },
+        {
+          _ref: 'www.vulture.com/_components/clay-paragraph/instances/cjeq5662h001b3i61rr8fffu1@published',
+        },
+        {
+          _ref: 'www.vulture.com/_components/article-sidebar/instances/foobar@published',
+          content: [
+            {
+              _ref: 'www.vulture.com/_components/blockquote/instances/pho@published',
+              componentVariation: 'blockquote'
+            }
+          ]
+        }
+      ],
+      variations = lib.getVariations('vulture');
+
+    lib.setDefaultVariation(testData, variations);
+
+    expect(testData[0]).to.deep.equal(expectedData[0]);
+    expect(testData[1]).to.deep.equal(expectedData[1]);
+    expect(testData[2]).to.deep.equal(expectedData[2]);
+    expect(testData[3]).to.deep.equal(expectedData[3]);
```

Comment:
can we simplify to `expect(testData).to.deep.equal(expectedData);` ?

Reply:
i tried doing that but mocha gets confused because its an array of OBJECTS and not an object?

Reply:
oh right of course